cluster/main.go

```diff
@@ -0,0 +1,192 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	ds "github.com/ipfs/go-datastore"
+	logging "github.com/ipfs/go-log"
+	ipfscluster "github.com/ipfs/ipfs-cluster"
+	"github.com/ipfs/ipfs-cluster/allocator/ascendalloc"
+	"github.com/ipfs/ipfs-cluster/allocator/descendalloc"
+	"github.com/ipfs/ipfs-cluster/config"
+	"github.com/ipfs/ipfs-cluster/consensus/crdt"
+	"github.com/ipfs/ipfs-cluster/informer/disk"
+	"github.com/ipfs/ipfs-cluster/informer/numpin"
+	"github.com/ipfs/ipfs-cluster/monitor/pubsubmon"
+	"github.com/ipfs/ipfs-cluster/observations"
+	"github.com/ipfs/ipfs-cluster/pintracker/maptracker"
+	"github.com/ipfs/ipfs-cluster/pintracker/stateless"
+	host "github.com/libp2p/go-libp2p-host"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+var log = logging.Logger("tex-cluster")
+
+func InitCluster(repoPath, listenAddr string) error {
+	cfgMgr, cfgs := makeClusterConfigs()
+	err := cfgMgr.Default()
+	if err != nil {
+		return err
+	}
+
+	if listenAddr != "" {
+		addr, err := ma.NewMultiaddr(listenAddr)
+		if err != nil {
+			return err
+		}
+		cfgs.ClusterCfg.ListenAddr = addr
+	}
+
+	return cfgMgr.SaveJSON(ConfigPath(repoPath))
+}
+
+func ConfigPath(repoPath string) string {
+	return filepath.Join(repoPath, "service.json")
+}
+
+func MakeAndLoadConfigs(repoPath string) (*config.Manager, *cfgs, error) {
+	cfgMgr, cfgs := makeClusterConfigs()
+	err := cfgMgr.LoadJSONFromFile(ConfigPath(repoPath))
+	if err != nil {
+		return nil, nil, err
+	}
+	return cfgMgr, cfgs, nil
+}
+
+func ParseBootstraps(addrs []string) ([]ma.Multiaddr, error) {
+	var parsed []ma.Multiaddr
+	for _, a := range addrs {
+		p, err := ma.NewMultiaddr(a)
+		if err != nil {
+			return nil, err
+		}
+		parsed = append(parsed, p)
+	}
+	return parsed, nil
+}
+
+func Bootstrap(ctx context.Context, cluster *ipfscluster.Cluster, cons ipfscluster.Consensus, bootstraps []ma.Multiaddr) {
+	for _, bstrap := range bootstraps {
+		log.Infof("Bootstrapping to %s", bstrap)
+		err := cluster.Join(ctx, bstrap)
+		if err != nil {
+			log.Errorf("bootstrap to %s failed: %s", bstrap, err)
+		} else {
+			for _, p := range cluster.Peers(ctx) {
+				err = cons.Trust(ctx, p.ID)
```

Comment:
I guess this is a +1 on auto-trusting peers you bootstrap to. So might get this done in Cluster (it's not yet there).

Reply:
Cool, maybe a config setting to auto-trust?

Reply:
Thinking about this. The bootstrap concept in cluster is part of the daemon binary (not of the cluster library itself), so while we can fix this there (https://github.com/ipfs/ipfs-cluster/issues/834), it won't help you.